packages/vite/src/node/optimizer/index.ts

```diff
@@ -253,7 +253,9 @@ export async function optimizeDeps(
         `Unable to parse dependency: ${id}. Trying again with a JSX transform.`
       )
       const transformed = await transformWithEsbuild(entryContent, filePath, {
-        loader: 'jsx'
+        loader: esbuildOptions.loader
+          ? esbuildOptions.loader[path.extname(filePath)]
+          : 'jsx'
```

Comment:
Should this be 

```js
loader: esbuildOptions.loader[path.extname(filePath)] || 'jsx'
```

instead? Looks like it could be undefined.

Reply:
OK, fixed.